djangobb_forum/static/djangobb_forum/scratchblocks/backpack.js

```diff
@@ -0,0 +1,108 @@
+/* Scratchblocks Backpack plugin
+* -----------------------------
+* 
+* Use the Scratch Backpack API to create 'add to backpack' buttons! In the
+* words of Dora the Explorer,
+*
+*   I'm the Backpack loaded up    /  With things and knick-knacks too!
+*   Anything that you might need  /  I got inside for you!
+*   Backpack, Backpack!
+*   Backpack, Backpack!
+*   Yeah!
+*
+* This plugin is MIT-licensed.
+*
+* -- Kartik, spring 2016
+*/
+~function(scratchblocks) {
+  // This is the API-level heavy lifting. When I say "API", I mean "what I
+  // glean from poking around cookies and tcpdump output", so, well, here's
+  // hoping the ST doesn't change anything anytime soon.
+
+  function addObjectToBackpack(obj, yes, no) {
+    // We're going to regex against the cookie because that's the One Best Way
+    // To Do It(tm).
+    var CSRFTOKEN = document.cookie?
+            (document.cookie.match(/csrftoken=(.*?);/) ?
+             document.cookie.match(/csrftoken=(.*?);/)[1] :
+             null) :
+            null;
+    var USER = window.Scratch && window.Scratch.INIT_DATA.LOGGED_IN_USER.model?
+           window.Scratch.INIT_DATA.LOGGED_IN_USER.model.username :
+           null;
+    if (!(CSRFTOKEN && USER)) {
+      return no('Not logged in.');
+    }
+    var x = new XMLHttpRequest();
+    x.open(
+      "GET",
+      "//scratch.mit.edu/internalapi/backpack/"+USER+"/get/",
```

Comment:
This URL, along with the one on line 52, should be relative. Fixing downstream. :)